scripts/generate_samples_list.dart

```diff
@@ -0,0 +1,63 @@
+// ignore_for_file: avoid_print
+
+import 'dart:convert';
+import 'dart:io';
+
+/// Run from arcgis-maps-sdk-flutter-samples root directory
+/// dart run scripts/generate_samples_list.dart
+void main() {
+  generateSamplesList();
+}
+
+/// Compiles all of the individual sample README.metadata.json files into a single JSON asset to feed into the Sample Viewer application.
+/// The output is in assets/generated_samples_list.json
+void generateSamplesList() {
+  Map<String, dynamic> samples = {};
+
+  print('Generating samples list....');
+
+  // get the lib/samples directory
+  final currentDirectory = Directory.current;
+  final samplesDirectory = Directory('${currentDirectory.path}/lib/samples');
+
+  if (samplesDirectory.existsSync()) {
+    // get all the individual sample directories
+    final individualSampleDirectories = samplesDirectory.listSync();
+
+    // iterate through each directory and get the README.metadata.json file for each sample
+    for (final directory in individualSampleDirectories) {
+      final directoryName = directory.path.split('/').last;
+      final metadataFile = (directory as Directory)
+          .listSync()
+          .firstWhere((file) => file.path.contains('metadata.json')) as File;
+      // get the json string for the sample
+      Map<String, dynamic> sampleJsonContent =
+          jsonDecode(metadataFile.readAsStringSync());
+      // add a key/value pair that is used by the Sample Viewer app
+      sampleJsonContent['key'] = directoryName;
+      // add the sample to the list of samples, using the directory name as the key
+      samples[directoryName] = sampleJsonContent;
+    }
+
+    // sort alphabetically
+    final sortedSamples = {
+      for (var k in samples.keys.toList()..sort()) k: samples[k]
+    };
+
+    // print list of samples to console
+    for (var key in sortedSamples.keys) {
+      print(key);
+    }
+    print('Total samples: ${sortedSamples.keys.length}');
+
+    // write the list of sorted samples to the generated_samples_list.json file
+    final assetsDirectory = Directory('${currentDirectory.path}/assets');
+    final samplesJsonFile =
+        File('${assetsDirectory.path}/generated_samples_list.json');
+    if (samplesJsonFile.existsSync()) {
```

Comment:
We should make this work whether or not `samplesJsonFile` already exists, since we're going to be blasting all new content on it in either case.

Reply:
My thinking here with these checks was more if say the script was run from the wrong directory (`Directory.current`), because then `samplesDirectory` could be the wrong path. I'll just throw here too if it can't find it per your other comment.

Reply:
How about if the check was "does the assets directory exist"? If there's an assets directory, we can safely assume that we are supposed to fill it with a `generated_samples_list.json` file, whether or not there's a `generated_samples_list.json` file already there. My concern is that I can easily see my future self deleting the generated file and getting surprised that the script no longer works.